virttest/env_process.py

```diff
@@ -1642,13 +1650,18 @@ def postprocess(test, params, env):
             logging.error(details)
 
     if params.get("storage_type") == "lvm":
-        try:
-            lvmdev = env.get_lvmdev("lvm_%s" % params["main_vm"])
-            lvmdev.cleanup()
-        except Exception as details:
-            err += "\nLVM cleanup: %s" % str(details).replace('\\n', '\n  ')
-            logging.error(details)
-        env.unregister_lvmdev("lvm_%s" % params["main_vm"])
+        for vm in params.objects("vms"):
+            vm_params = params.object_params(vm)
+            for image in vm_params.objects("images"):
```

Comment:
Without the abstraction of the pool, it is hard to make the clean-up procedure right, let's consider the following situation:

* there are more than one image (e.g. stg1 stg2)
* vg is dynamically created (requested by stg1)
* one of the image would not be removed after test (`remove_image = no`)

with the current implementation, it would remove the vg at here and then get failure.

Reply:
if `remove_image = no`, this cleanup process won't remove the `vg` created since the real parameter to control this behavior is `force_remove_image`, which is default `no`.

Reply:
Sorry, the example may not be correct (I mean the configuration part in the above comment), but the situation does exist, it is possible to construct a configuration which can trigger the failure I mentioned.

Reply:
From my perspective, supporting host devices is the most important point for the entire requirement: we have to ensure that images can be created, removed, backed up, restored ... with using host's blocks. Then the management layer of LVM could be had as an additional functionality.

Reply:
I've drafted out the `supporting host lvm device` functionality in this [path] (https://github.com/avocado-framework/avocado-vt/pull/2448) without lvm management, please take a look.